src/pngtocss.c

```diff
@@ -109,47 +102,95 @@ static void usage_info()
 	fprintf(stderr, "Usage: pngtocss <image1.png> <image2.png> ...\n");
 }
 
-static rgb byte2rgb(png_bytep triad)
+static rgba byte2rgba(png_bytep quartet)
 {
-	rgb c;
+	rgba c;
 
-	c.r = triad[0];
-	c.g = triad[1];
-	c.b = triad[2];
+	/* Pixels with an alpha value of 0 have bogus color values in the buffer, so
+	 * we are normalizing them to 0.
+	 */
+	if (quartet[3] == 0) {
+		c.r = 0;
+		c.g = 0;
+		c.b = 0;
+	}
+	else {
+		c.r = quartet[0];
+		c.g = quartet[1];
+		c.b = quartet[2];
+	}
+
+	c.a = quartet[3];
 
 	c.pos = -1;
 
 	return c;
 }
 
-static rgb getpixel(const bmp *image, png_uint_32 x, png_uint_32 y)
+static rgba getpixel(const image *image, png_uint_32 x, png_uint_32 y)
 {
-	return byte2rgb(image->data + (y*image->width*image->pxbytes) + x*image->pxbytes);
+	return byte2rgba(image->buffer +
+		(y*image->image.width*image->pixel_bytes) +
+		x*image->pixel_bytes);
 }
 
-static rgb rgb_avg(rgb a, rgb b)
+static rgba rgba_avg(rgba a, rgba b)
 {
-	rgb c;
+	rgba c;
+
+	/* Pixels with an alpha value of 0 will be black, which messes up the
+	 * averaging. To circumvent this, the black pixel will take on the color
+	 * values of the pixel it is being averaged with.
+	 */
+
+	if(a.a == 0) {
+		a.r = b.r;
+		a.g = b.g;
+		a.b = b.b;
+	}
+
+	if(b.a == 0) {
+		b.r = a.r;
+		b.g = a.g;
+		b.b = a.b;
+	}
 
 	c.r = (a.r+b.r)/2;
 	c.g = (a.g+b.g)/2;
 	c.b = (a.b+b.b)/2;
+	c.a = (a.a+b.a)/2;
 
 	return c;
 }
 
-static int rgb_equal(rgb a, rgb b)
+static int rgba_equal(const rgba a, const rgba b)
 {
-	/* Match with a tolerance of 2 */
-	if(a.r <= b.r+2 && a.r >= b.r-2
-	&& a.g <= b.g+2 && a.g >= b.g-2
-       	&& a.b <= b.b+2 && a.b >= b.b-2) {
+	const int tolerance = 2;
+	if(a.a == 0 && b.a == 0)
 		return 1;
-	}
+	/* Match with a tolerance */
+	else if(a.r <= b.r+tolerance && a.r >= b.r-tolerance &&
+			a.g <= b.g+tolerance && a.g >= b.g-tolerance &&
+			a.b <= b.b+tolerance && a.b >= b.b-tolerance &&
+			a.a <= b.a+tolerance && a.a >= b.a-tolerance)
+		return 1;
+	
 	return 0;
 }
 
-static void print_rgb(rgb c)
+static double decimal2ratio(int a)
+{
+	char buffer[256];
+	sprintf(buffer, "%.2f", a/255.0);
+	return atof(buffer);
+}
+
+static void print_rgba(rgba c)
+{
+	printf("rgba(%i,%i,%i,%.3g)", c.r, c.g, c.b, decimal2ratio(c.a));
```

Comment:
if we're using `%.2f` above, then does `%.3g` make sense here?  Should we just get rid of `decimal2ratio` and use `%.2f` here with the appropriate `/255.0`?